docs/real-time-analytics/discounting/process.sql

```diff
@@ -0,0 +1,64 @@
+SET 'auto.offset.reset' = 'earliest';
+
+-- Create Discount Code Table
+CREATE TABLE DISCOUNT_CODES (
+    CODE VARCHAR PRIMARY KEY, 
+    PERCENTAGE DOUBLE
+) WITH (
+    KAFKA_TOPIC = 'discount_codes',
+    VALUE_FORMAT = 'JSON',
+    PARTITIONS = 6
+);
+
+-- Create Order Stream
+CREATE STREAM ORDER_STREAM (
+	ID BIGINT, 
+	ORDER_TIME TIMESTAMP, 
+	CUSTOMER_ID BIGINT, 
+	ITEM_CT INT, 
+	ORDER_SUBTOTAL DOUBLE, 
+	DISCOUNT_CODE VARCHAR
+) WITH (
+  KAFKA_TOPIC = 'orders',
+  VALUE_FORMAT = 'JSON',
+  PARTITIONS = 6
+);
+
+
+-- Application logic
+
+-- Enrich the orders stream with discount details.
+-- Compute the discount amount as well as order total.
+CREATE STREAM ORDERS_ENRICHED WITH (
+	KAFKA_TOPIC = 'orders_enriched',
+    VALUE_FORMAT = 'JSON',
+    PARTITIONS = 6
+) AS 
+SELECT 
+	OS.ID AS ORDER_ID,
+	OS.DISCOUNT_CODE,
+	DISCOUNT_CODES.PERCENTAGE,
+	OS.ITEM_CT,
+	OS.ORDER_SUBTOTAL, 
+	OS.ORDER_SUBTOTAL * (100 - DISCOUNT_CODES.PERCENTAGE) / 100.0 AS ORDER_TOTAL,
+	OS.ORDER_SUBTOTAL * (DISCOUNT_CODES.PERCENTAGE / 100.0) AS DISCOUNT_TOTAL
+FROM ORDER_STREAM OS 
+INNER JOIN DISCOUNT_CODES
+ON OS.DISCOUNT_CODE = DISCOUNT_CODES.CODE
+EMIT CHANGES;
+
+-- Compute aggregates based on discount percentage.
+CREATE TABLE DISCOUNT_ANALYSIS WITH (
+	KAFKA_TOPIC = 'discount_analysis',
+    VALUE_FORMAT = 'JSON',
+    PARTITIONS = 6
+) AS 
+SELECT
+	PERCENTAGE,
+	ROUND(AVG(ITEM_CT), 0) AS AVG_ITEM_CT,
+	ROUND(AVG(ORDER_SUBTOTAL), 2) AS AVG_ORDER_SUBTOTAL,
+	ROUND(AVG(ORDER_TOTAL), 2) AS AVG_ORDER_TOTAL,
+	ROUND(AVG(DISCOUNT_TOTAL), 2) AS AVG_DISCOUNT_TOTAL
+FROM ORDERS_ENRICHED
+GROUP BY PERCENTAGE
```

Comment:
It would be more efficient to merge both queries into a single one. -- We had similar discussion on other recipes: it's always a trade-off to show most efficient vs more readable queries...

Reply:
I personally feel that, in the case of recipes, we should be opting for readability so that users understand exactly what's happening in each step... Here we have enrichment _then_ analysis. But I'm happy to merge these for consistency. Let me know.

Reply:
Maybe @ybyzek can help.

We did rewrite a few recipes and merged queries. But I guess it always depends and is a case-by-case decision. Don't have too strong of an opinion. -- The main argument for merging is, that people should learn how to write efficient code early (also given the current 20 query limit in cloud; to be fair we plan to increase it soon, it might become less pressing). Of course, there is some balance to keep...